src/functions/services/logging/logger.test.ts

```diff
@@ -0,0 +1,125 @@
+import { LogAttributes } from "@aws-lambda-powertools/logger/lib/cjs/types/Log";
+import { ILoggerAdapter, Logger } from "./logger";
+import { LogMessage, RegisteredLogMessages } from "./types";
+import { Context } from "aws-lambda";
+
+describe("Logger", () => {
```

Comment:
Haven't got to the logger file itself yet but was expecting to see a test to say if we try and log a message that isn't registered it throws an error. Do we want this functionality/test in place or a future enhancement, or have I misunderstood the value of registered log messages?

Reply:
It shouldn't be possible given the type system. Give it a go!

Reply:
Nice I like it. I still wonder whether there should be a unit test to show it won't work.. might be overkill but my thinking:
- our unit tests are very good documentation of what things should be doing
- Edge cases or unexpected inputs that TS isn't protecting against
- If for any reason we did create a runtime error (which I think is highly unlikely) the tests will catch this

Appreciate typescript will throw a type error when writing the test but I found at least one way to get around it:

 it('should throw an error when logging an unregistered message', () => {
        // @ts-expect-error: Testing a case that should not be allowed by TypeScript
        expect(() => logger.log('info', 'This should fail!!')).toThrow('Message type info is not registered');
        
Happy tho if you think this is overkill or if a logic change seems easier

Reply:
Thanks @jamesamooney. Explicitly overriding typescript's inbuilt linting isn't a valid scenario for me. 

For quality here, we use a combination of types and unit tests. 

I'm happy to close this without changing.